app/forms/legal_aid_applications/used_multiple_delegated_functions_form.rb

```diff
@@ -0,0 +1,166 @@
+module LegalAidApplications
+  class UsedMultipleDelegatedFunctionsForm # rubocop:disable Metrics/ClassLength
+    include ActiveModel::Model
+
+    validate :validate_nothing_selected,
+             :validate_proceeding_selected,
+             :validate_proceeding_dates
+
+    attr_accessor :draft,
+                  :proceeding_types_by_name,
+                  :none_selected
+
+    class << self
+      def call(proceeding_types_by_name)
+        populate_attr_accessors(proceeding_types_by_name.map(&:name))
+        new({ proceeding_types_by_name: proceeding_types_by_name })
+      end
+
+      def populate_attr_accessors(proceeding_names)
+        proceeding_names.each do |name|
+          attr_accessor :"#{name}",
+                        :"#{name}_used_delegated_functions_on_1i",
+                        :"#{name}_used_delegated_functions_on_2i",
+                        :"#{name}_used_delegated_functions_on_3i",
+                        :"#{name}_used_delegated_functions_on"
+        end
+      end
+    end
+
+    def initialize(*args)
+      super
+      proceeding_types_by_name.each do |proceeding|
+        date = proceeding.application_proceeding_type.used_delegated_functions_on
+        next unless date
+
+        __send__("#{proceeding.name}=", 'true')
+        __send__("#{proceeding.name}_used_delegated_functions_on=", date)
+      end
+    end
+
+    def save(params)
+      update_proceeding_attributes(params)
+      update_populated_dates(params)
+
+      return false unless valid? || draft_nothing_selected?
+
+      save_proceeding_records
+    end
+
+    def earliest_delegated_functions
```

Comment:
Similar to above,  `proceeding_with_earliest_delegated_functions` might be a clearer name.